src/Kestrel.Core/Internal/Http/Http1MessageBody.cs

```diff
@@ -125,19 +124,55 @@ public override Task StopAsync()
             return _pumpTask;
         }
 
-        protected override async Task OnConsumeAsync()
+        protected override Task OnConsumeAsync()
         {
-            _context.TimeoutControl.SetTimeout(Constants.RequestBodyDrainTimeout.Ticks, TimeoutAction.SendTimeoutResponse);
+            try
+            {
+                if (_context.RequestBodyPipe.Reader.TryRead(out var readResult))
+                {
+                    _context.RequestBodyPipe.Reader.AdvanceTo(readResult.Buffer.End);
+
+                    if (readResult.IsCompleted)
+                    {
+                        return Task.CompletedTask;
+                    }
+                }
+            }
+            catch (BadHttpRequestException ex)
```

Comment:
How would this get thrown from the code above? We're not calling anything that would result in this exception

Reply:
Wait, I liked, hmm it looks like TryRead does throw if the writer completed with an error? That right @pakrym ?

Reply:
[It does.](https://github.com/dotnet/corefx/blob/0b9a376b08331692ce68ebffb38e3bcf3c05245b/src/System.IO.Pipelines/src/System/IO/Pipelines/Pipe.cs#L760)